cypress/integration/dashboard/dashboard_filtering_spec.js

```diff
@@ -0,0 +1,315 @@
+      cy.checkElementDoesNotExist('svg > g > g.arcs > path.slice')
+    })
+
+    it('area, bar and heatmap charts filtered', () => {
+      // Check that none of the charts are filled with data
+      cy.checkElementDoesNotExist('svg > g > g.series')
+    })
+
+    it('data tables are filtered', () => {
+      // Check that none of the data tables are filled with data
+      cy.checkElementDoesNotExist('[data-test-subj="paginated-table-body"] [data-cell-content]')
+    })
+
+    it('goal and guages are filtered', () => {
+      // Goal label should be 0, gauge label should be 0%
+      cy.checkElementContainsValue('svg > g > g > text.chart-label', 2, /^0%?$/)
+    })
+
+    it('metric value shows no data', () => {
+      // The metrics should show '-'
+      cy.checkElementContainsValue('.mtrVis__value', 1, /^ - $/)
+    })
+
+    it('tag cloud values are filtered', () => {
+      cy.checkElementComponentDoesNotExist('[data-test-subj="tagCloudVisualization"]', 'svg > g > text')
+    })
+
+    it('tsvb metric is filtered', () => {
+      cy.checkElementContainsValue('[data-test-subj="tsvbMetricValue"]', 1, /^0 custom template$/)
+    })
+
+    it('tsvb top n is filtered', () => {
+      cy.checkElementContainsValue('[data-test-subj="tsvbTopNValue"]', 2, /^0$/)
+    })
+
+    it('saved search is filtered', () => {
+      cy.get('[data-test-subj="docTableExpandToggleColumn"]').should('not.exist')
+    })
+
+    // TODO: Uncomment once https://github.com/elastic/kibana/issues/22561 is fixed
+    // it('timeline is filtered', async () => {
+    //   await dashboardExpect.timelineLegendCount(0)
+    // })
+
+    it('vega is filtered', () => {
+      cy.get('.vgaVis__view text').each(($el, index, $list) => {
+        cy.get($el).should('not.contain', /^5,000$/)
+      })
+    })
+  })
+
+  describe('disabling a filter unfilters the data on', () => {
+    before(() => {
+      // TO DO: create delete filter helper function
+      cy.get('[data-test-subj="filter filter-enabled filter-key-bytes filter-value-12,345,678 filter-unpinned "]').click()
+      cy.get('[data-test-subj="deleteFilter"]').click()
+      cy.get('[data-test-subj="filter filter-enabled filter-key-bytes filter-value-12,345,678 filter-unpinned "]').should('not.exist')
+    })
+
+    it('pie charts', () => {
+      // Check that there are 5 slice in the pie charts
+      cy.checkElementExists('svg > g > g.arcs > path.slice', 5)
+    })
+
+    it('area, bar and heatmap charts', () => {
+      // Check that there are 3 charts
+      cy.checkElementExists('svg > g > g.series', 3)
+    })
+
+    it('data tables', () => {
+      // Check that there are 10 table rows
+      cy.checkElementExists('[data-test-subj="paginated-table-body"] [data-cell-content]', 10)
+    })
+
+    it('goal and guages', () => {
+      // Goal label should be 7,544, gauge label should be 39.958%%
+      // Inconsistency: original code wants 7,544, current UI wants 7,565
```

Comment:
Do you know why we have inconsistency?

Reply:
I am personally unsure where the inconsistency came from for the goal label. When running the tests recently, the goal label went back to correctly displaying "7,544" and have continued to do so. My best guess would be that there may have been a data importing inconsistency that occurred when experimenting with loading and unloading test environment data.